dnssec: validate DS records

japaric opened the pull request:
this PR ports 4 false positives scenarios from `ede-dot-com` into `conformance-tests`. Adds 3 DS checks to the DNSSEC validation code which fix all the newly added conformance tests.

Observably, this only fixes 2 `ede-dot-com` tests because the other 2 scenarios still fail due to  #2395 

~~this PR depends on #2392 so opening in draft state until that is merged~~

djc commented on crates/proto/src/xfer/dnssec_dns_handle.rs:
```diff
@@ -449,6 +449,16 @@ where
         for r in ds_records.iter() {
+            if r.data().algorithm() != key_rdata.algorithm() {
+                debug!(
+                    "skipping DS record due algorithm mismatch: ({}, {})",
```
"due [to]"? Maybe also log the algorithms here?

djc commented on crates/proto/src/xfer/dnssec_dns_handle.rs:
```diff
@@ -449,6 +449,16 @@ where
+                debug!(
+                    "skipping DS record due algorithm mismatch: ({}, {})",
+                    r.name(),
+                    r.data()
```
Nit: and a trailing comma here, rustfmt willing?

marcus0x62 commented on crates/proto/src/xfer/dnssec_dns_handle.rs:
```diff
@@ -448,7 +448,31 @@ where
+
         for r in ds_records.iter() {
+            if r.data().algorithm() != key_rdata.algorithm() {
+                debug!(
```
Suggestion: make this and the other call at line 467 trace! to keep the noise down in the log file (these are going to fire on pretty much every query for a DNSSEC zone), and also nuke the trace! call at line 489; it is superfluous given the debug! call at line 480.

japaric commented:
addressed review comments and updated the newly added tests to adhere to the changes in the `dns_test` API ( #2381 ). note that this PR will fail the `conformance` job until #2405 is merged

djc commented:
Just needs another rebase to pull in #2405, I think?
